src/state/saga/auth.saga.jsx

```diff
@@ -0,0 +1,46 @@
+import axios from "axios";
+import {all, call, takeLatest} from '@redux-saga/core/effects';
+
+import {setAuthTokenCookie} from "../../tool/auth.util";
+import {getAuthorizeRequest} from "../tool/api/auth.endpoint";
+import {RESPONSE_STATUS} from "../../tool/constant";
+import {GET_AUTH_TOKEN} from "../action/auth.actions";
+
+export function* performAuthToken({payload, navigate, onError, onComplete}) {
+
+    try {
+        const [endpoint, requestOptions] = getAuthorizeRequest(payload);
+
+        const {data} = yield call(axios, endpoint, requestOptions);
+
+        const token = data.token;
+
+        yield call(setAuthTokenCookie, token);
+
+        //Refresh current page.
+        yield call(navigate, 0);
```

Comment:
Why refresh the page? Why not route to a new page?

Reply:
The reason I did this is because if someone navigates to e.g. _**/budget**_ and they aren't authenticated it shows the login page. Once they are logged in it refreshes the page they have requested.